core/src/autogluon/core/hpo/ray_hpo.py

```diff
@@ -212,10 +214,8 @@ def run(
     scheduler = _get_scheduler(hyperparameter_tune_kwargs, supported_schedulers=ray_tune_adapter.get_supported_schedulers())
     search_space = _convert_search_space(search_space)
 
-    if ray.is_initialized():
-        # shutdown to reinitialize resources because different model might require different total resources
-        ray.shutdown()
-    ray.init(log_to_driver=False, **total_resources)
+    if not ray.is_initialized():
+        ray.init(log_to_driver=False, **total_resources)
```

Comment:
Does this leave any artifact if an exception is raised and ray is not shutdown, or does it auto shutdown?

Reply:
This is a good question and I don't know the answer as I don't find ray's documentation on this. However, when the process calling ray.init() terminates, the Ray runtime will also terminate.

Also, this is just the ray runtime not the cluster, so shouldn't be a big deal